### PR TITLE
[SYSDM] User Profiles fix of OnDestroy (CORE-16921)

### PR DESCRIPTION
**Pull request by KyleKatarn31**
## Purpose

_ User Profiles window incorrectly release Heap zone in its OnDestroy() function, leading to undue closure of its parent Window

JIRA issue: [CORE-16921](https://jira.reactos.org/browse/CORE-16921)

## Proposed changes

_ Correct initialization of "Item" variable before using it
_ Protection against invalid Handle usage
_ Consistency of variable naming accross the window

### CODE REVIEW

**Comment by katahiromz** [comment]
Using `LVN_DELETEITEM` notification is the better way.

**Comment by KyleKatarn31** [comment]
> Using `LVN_DELETEITEM` notification is the better way.

I understand. However, i'd prefer to "maintain" the current code design principle and "fix what is broken" rather than redesigning with WM_NOTIFY + LVN_DELETEITEM.

Would you accept to keep current structure (fixed as propose), or would you only accept a redesigned code based on WM_NOTIFY with LVN_DELETEITEM ?

File: dll/cpl/sysdm/userprofile.c [hunk 1]
```diff
@@ -645,6 +645,7 @@ UpdateButtonState(
         iSelected = ListView_GetNextItem(hwndListView, -1, LVNI_SELECTED);
         if (iSelected != -1)
         {
+            ZeroMemory(&Item, sizeof(LVITEM));
             Item.mask = LVIF_PARAM;
             Item.iItem = iSelected;
             Item.iSubItem = 0;
```
Comment (SergeGautherie) [inline dll/cpl/sysdm/userprofile.c:651]: This line becomes redundant.

Comment (KyleKatarn31) [reply dll/cpl/sysdm/userprofile.c:651]: Why redundant ? It allows the proper initialization to 0 of all fields but the ones used for search. Am I wrong ?

Comment (KyleKatarn31) [reply dll/cpl/sysdm/userprofile.c:651]: Oups, my mistake.
Indeed "Item.iSubItem = 0;" is useless after a ZeroMemory. This code pattern is used 4 times in this file (2 times already like this prior to the change).
Would you suggest that I remove the line setting iSubItem to 0 all 4 times ?

Comment (KyleKatarn31) [reply dll/cpl/sysdm/userprofile.c:651]: Oups, my mistake.
Indeed "Item.iSubItem = 0;" is useless after a ZeroMemory. This code pattern is used 4 times in this file (2 times already like this prior to the change).
Would you suggest that I remove the line setting iSubItem to 0 all 4 times ?

**Comment by KyleKatarn31** [comment]
@SergeGautherie : Now hat you comments are integrated, OK for approval ?

**Comment by KyleKatarn31** [comment]
@katahiromz @sdever : New implementation based on LVN_DELETEITEM as suggested.

Tested OK

If OK for you, would you please approve and merge ?

**Comment by KyleKatarn31** [comment]
Thanks !

**Pull request merged**